R/document.R

```diff
@@ -0,0 +1,126 @@
+#'
+#' @details
+#' `title`, `type`, `space_key`, `parent_id`, `toc`, `toc_depth`, `update`, and
+#' `use_original_size` can be specified as `confluence_settings` item in the
+#' front-matter of the Rmd file to knit. The arguments of
+#' `confl_create_post_from_Rmd()` overwrite these settings if provided.
+#'
+#' @rdname confluence_document
+#'
+#' @export
+confluence_document <- function(interactive = FALSE,
+                                title = NULL,
+                                # Use snake case for user-facing functions and use the actual API parameter name
+                                # in camel case for simple binding functions.
+                                space_key = NULL,
+                                type = NULL,
+                                parent_id = NULL,
+                                toc = NULL,
+                                toc_depth = NULL,
+                                supported_syntax_highlighting = getOption("conflr_supported_syntax_highlighting"),
+                                update = NULL,
+                                use_original_size = NULL) {
+  # This will be refered in post_processor()
+  confluence_settings_from_args <- list(
+    title = title,
+    space_key = space_key,
+    type = type,
+    parent_id = parent_id,
+    toc = toc,
+    toc_depth = toc_depth,
+    supported_syntax_highlighting = supported_syntax_highlighting,
+    update = update,
+    use_original_size = use_original_size
+  )
+
+  format <- rmarkdown::md_document(
+    variant = "commonmark",
+    pandoc_args = "--wrap=none",
+    md_extensions = "-tex_math_single_backslash-tex_math_dollars-raw_tex",
+    preserve_yaml = FALSE
+  )
+
+  format$post_processor <- function(front_matter, input_file, output_file, clean, verbose) {
+    # For backward-compatibility
+    if (has_name(front_matter, "confluence_settings")) {
+      warn(paste0("Set options via `confluence_settings` front-matter is deprecated.\n",
+                  "Please use `confluence_document` instead."))
+
+      confluence_settings <- front_matter$confluence_settings
+
+      confluence_settings <- purrr::list_modify(
+        confluence_settings,
+        !!!purrr::compact(confluence_settings_from_args)
+      )
+    } else {
+      confluence_settings <- confluence_settings_from_args
+    }
+
+    # title can be specified as a seperate item on front matter
+    confluence_settings$title <- confluence_settings$title %||% front_matter$title
+
+    md_text <- read_utf8(output_file)
+
+    # Replace <ac:...> and <ri:...> because they are not recognized as proper tags
+    # by commonmark and accordingly get escaped. We need to replace the namespace
+    # to bypass the unwanted conversions. The tags will be restored later in
+    # confl_upload().
+    md_text <- mark_confluence_namespaces(md_text)
+
+    html_text <- commonmark::markdown_html(md_text)
+
+    imgs <- extract_image_paths(html_text)
+    imgs_realpath <- curl::curl_unescape(imgs)
```

Comment:
Note to self: We no longer need to convert `imgs` to absolute paths because we are in the directory `rmarkdown::render()` is using.